src/react.tsx

```diff
@@ -28,12 +35,20 @@ export function useRemixI18Next(locale: string) {
       )
   );
 
-  useMemo(() => {
+  const handleLocaleUpdate = useCallback(() => {
     i18next.changeLanguage(locale);
     for (let [namespace, messages] of Object.entries(namespaces)) {
       i18next.addResourceBundle(locale, namespace, messages);
     }
   }, [i18next, namespaces, locale]);
+
+  useMemo(() => {
+    handleLocaleUpdate();
+  }, []);
+
+  useEffect(() => {
+    handleLocaleUpdate();
+  }, [handleLocaleUpdate]);
```

Comment:
calls `handleLocaleUpdate` on subsequent renders but only when the identity of `handleLocaleUpdate` changes.